### PR TITLE
Verbum: change reply to comment where needed

### PR DESCRIPTION
**Pull request by heavyweight**
This PR fixes the wording around comments.
**Comment**
1. When posting a comment the submit button should be `Comment`.
2. The textarea placeholder should be `Write a comment...`.
3. Depending on the comment auth config we should either display: `Log in to leave a comment.` or `Leave a comment. (log in optional)`
4. The title above the form should say: `Leave a Comment`

**Reply**
1. When posting a reply the submit button should be `Reply`.
2. The textarea placeholder should be `Write a reply...`.
3. Depending on the comment auth config we should either display: `Log in to leave a reply.` or `Leave a reply. (log in optional)`

This PR fixes the e2e tests, and adds a command to run them `e2e-tests`.

Fixes #
136-gh-Automattic/verbum

## Proposed changes:

* Introduce `commentParent` state to keep track if we are replying or just commenting.
* Introduce new hook to watch for form mutations and update the new state.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Checkout this PR
* Run the e2e tests and ensure that everything is ok
* Verify all the string changes in the PR description

### CODE REVIEW

File: projects/packages/jetpack-mu-wpcom/package.json [hunk 1]
```diff
@@ -21,7 +21,8 @@
 		"build-production-js": "NODE_ENV=production BABEL_ENV=production pnpm build-js",
 		"lint": "eslint --ext .js,.jsx,.cjs,.mjs,.ts,.tsx .",
 		"clean": "rm -rf src/build/",
-		"watch": "pnpm run build-production-js && pnpm webpack watch"
+		"watch": "pnpm run build-production-js && pnpm webpack watch",
+		"e2e-tests": "playwright test --config src/features/verbum-comments/playwright.config.ts"
```
Comment (agrullon95) [inline projects/packages/jetpack-mu-wpcom/package.json:25]: Im getting a "No tests found" error when I run `pnpm run e2e-tests` from the jetpack-mu-wpcom root folder.

File: projects/packages/jetpack-mu-wpcom/src/features/verbum-comments/README.md [hunk 2]
```diff
@@ -115,6 +115,23 @@ On Atomic Sites: Sync the Verbum code changes to your sandbox and sandbox `jetpa
...
+
+1. Please sandbox the following sites before running the tests
+	- jetpack.wordpress.com
+	- e2esiteopencommentstoeveryone.wordpress.com
+	- e2ecommentauthormustfilloutnameandemail.wordpress.com
+	- e2eusersmustberegisteredandloggedintocomment.wordpress.com
+
+2. Run `npx playwright install` to install the browsers needed.
+
+The tests live in /tests folder. To run them, you can run `npm run e2e-tests`.
```
Comment (agrullon95) [inline projects/packages/jetpack-mu-wpcom/src/features/verbum-comments/README.md]: `npm run e2e-tests` should be `pnpm run e2e-tests`

File: projects/packages/jetpack-mu-wpcom/src/features/verbum-comments/class-verbum-comments.php [hunk 3]
```diff
@@ -272,7 +276,7 @@ public function comment_form_defaults( $args ) {
 				'logged_in_as'         => '',
 				'comment_notes_before' => '',
 				'comment_notes_after'  => '',
-				'title_reply'          => __( 'Leave a Reply', 'jetpack-mu-wpcom' ),
+				'title_reply'          => __( 'Leave a comment', 'jetpack-mu-wpcom' ),
```
Comment (agrullon95) [inline projects/packages/jetpack-mu-wpcom/src/features/verbum-comments/class-verbum-comments.php:280]: I updated this to `Leave a comment` according to @crisbusquets's comment: p1706631945152779/1706626540.417219-slack-C02T4NVL4JJ

**Comment by agrullon95** [comment]
Good work! Left a few small comments.

### Testing in progress:

**Comment**

- [x] When posting a comment the submit button should be Comment.
- [x] The textarea placeholder should be Write a comment....
- [ ] Depending on the comment auth config we should either display: Log in to leave a comment. or Leave a comment. (log in optional)
- [x] The title above the form should say: Leave a comment

**Reply**

- [x] When posting a reply the submit button should be Reply.
- [x] The textarea placeholder should be Write a reply....
- [ ] Depending on the comment auth config we should either display: Log in to leave a reply. or Leave a reply. (log in optional)
- [ ] This PR fixes the e2e tests, and adds a command to run them e2e-tests. **(Im unable to get the tests to run using `pnpm run e2e-tests` from root directory)**

File: projects/packages/jetpack-mu-wpcom/src/features/verbum-comments/src/components/logged-out.tsx [hunk 4]
```diff
@@ -11,8 +12,12 @@ interface LoggedOutProps {
...
-	const defaultText = translate( 'Log in to leave a reply.' );
-	const optionalText = translate( 'Leave a reply. (log in optional)' );
+	const defaultText = commentParent.value
+		? translate( 'Log in to leave a reply.' )
+		: translate( 'Log in to leave a comment.' );
+	const optionalText = commentParent.value
+		? translate( 'Leave a reply. (log in optional)' )
+		: translate( 'Leave a comment. (log in optional)' );
 	const nameAndEmailRequired = translate(
 		'Log in or provide your name and email to leave a reply.'
```
Comment (agrullon95) [inline projects/packages/jetpack-mu-wpcom/src/features/verbum-comments/src/components/logged-out.tsx]: We should update this text as well:
`Log in or provide your name and email to leave a reply.`
`Log in or provide your name and email to leave a comment.`

WDYT?

Comment (crisbusquets) [reply projects/packages/jetpack-mu-wpcom/src/features/verbum-comments/src/components/logged-out.tsx]: Good catch. Works for me!

File: projects/packages/jetpack-mu-wpcom/package.json (hunk 1 above)
Comment (heavyweight) [reply projects/packages/jetpack-mu-wpcom/package.json:25]: Screenshot 2024-01-31 at 10 00 46
Its working for me.
Can you double-check the folder and node version?

**Review by escapemanuele — CHANGES_REQUESTED** [review]
In simple sites all is working great!

It's not working properly in Atomic for me though.

This is a normal comment:

image

This is a reply:

image

**Comment by heavyweight** [comment]
> In simple sites all is working great!
> 
> It's not working properly in Atomic for me though.
> 
> This is a normal comment:
> 
> image
> 
> This is a reply:
> 
> image

Pushed a fix for this

**Comment by escapemanuele** [comment]
Much better, only thing is that I still see `Leave a Reply` in Atomic comment. Reply works great ✌️ 

image

**Review by escapemanuele — APPROVED** [review]
Tested it and also tested the new E2E tests, great work!

**Pull request merged**